pkg/skaffold/deploy/status_check.go

```diff
@@ -188,22 +227,22 @@ func getDeadline(d int) time.Duration {
 	return defaultStatusCheckDeadline
 }
 
-func printStatusCheckSummary(out io.Writer, r *resource.Deployment, c counter) {
+func (s statusChecker) printStatusCheckSummary(out io.Writer, r *resource.Deployment, c counter) {
 	ae := r.Status().ActionableError()
-	if ae.ErrCode == proto.StatusCode_STATUSCHECK_CONTEXT_CANCELLED {
-		// Don't print the status summary if the user ctrl-C
+	if r.StatusCode() == proto.StatusCode_STATUSCHECK_CONTEXT_CANCELLED {
+		// Don't print the status summary if the user ctrl-C or
+		// another deployment failed
```

Comment:
If we can check the context for cancellation, then we should not need this code.

Reply:
This is still needed. It was added in #3892.

When fetching deployment status, we run `kubectl rollout status` command in background. 
If user hits Cntrl-C, the command is cancelled and deployments status is updated as "context cancelled"

here is an example on how it looks on master 
```
Waiting for deployments to stabilize...
 - deployment/nodejs-guestbook-frontend: waiting for rollout to finish: 0 of 1 updated replicas are available...
    - pod/nodejs-guestbook-frontend-5f954f8b9d-h692w: creating container frontend
 - deployment/nodejs-guestbook-mongodb: waiting for rollout to finish: 0 of 1 updated replicas are available...
    - pod/nodejs-guestbook-mongodb-59b8b89f4c-jkmft: creating container mongo
^C - deployment/nodejs-guestbook-backend failed. [2/3 deployment(s) still pending] Error: could not stabilize within 2m0s: context canceled.
 - deployment/nodejs-guestbook-mongodb: could not stabilize within 2m0s: context canceled    - pod/nodejs-guestbook-mongodb-59b8b89f4c-jkmft: creating container mongo
 - deployment/nodejs-guestbook-mongodb failed. [1/3 deployment(s) still pending] Error: could not stabilize within 2m0s: context canceled.
 - deployment/nodejs-guestbook-frontend: could not stabilize within 2m0s: context canceled    - pod/nodejs-guestbook-frontend-5f954f8b9d-h692w: creating container frontend
 - deployment/nodejs-guestbook-frontend failed. Error: could not stabilize within 2m0s: context canceled
```
You can see ` - deployment/nodejs-guestbook-frontend failed. Error: could not stabilize within 2m0s: context canceled`

Reply:
IMHO these _context canceled_ items shouldn't be shown.